File: LeetCode/C++/trees-binary-tree-level-order-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
private:
    
public:    
    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int>> tree;
        
        if (root) {
            queue<TreeNode*> q;
            q.push(root);
        
            while(!q.empty()) {
                vector<int> level;
                int q_size = q.size();

                for(int i=0; i < q_size; i++) {
                    TreeNode *node = q.front(); 
                    level.push_back(node->val);

                    if(node->left)
                        q.push(node->left);
                    if(node->right)
                        q.push(node->right);
                    q.pop();
                }
                tree.push_back(level);
            }
        }
        
        return tree;    
    }
};

File: LeetCode/Python/trees-binary-tree-level-order-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        tree = list()
        
        if root:
            q = list()
            q.append(root)
            
            while q:
                level = list()
                q_size = len(q)
                
                for i in range(q_size):
                    node = q.pop(0)
                    level.append(node.val)
                    
                    if node.left:
                        q.append(node.left)
                    if node.right:
                        q.append(node.right)
                
                tree.append(level)
        
        return tree
